sites/partners/src/components/listings/ListingFormActions.tsx

```diff
@@ -195,60 +200,249 @@ const ListingFormActions = ({
+                await router.push(`/listings/${result.id}`)
+              }
+            } catch (err) {
+              setSiteAlertMessage("errors.somethingWentWrong", "warn")
+            }
+          }}
+        >
+          {t("listings.approval.approveAndPublish")}
+        </Button>
+      </GridCell>
+    )
 
-    // read-only form
-    if (type === "details") {
-      elements.push(editFromDetailButton)
-      elements.push(previewButton)
+    const requestChangesButton = (
+      <GridCell key="btn-request-changes">
+        <Button
+          id="submitButton"
+          styleType={AppearanceStyleType.alert}
+          border={AppearanceBorderType.outlined}
+          type="button"
+          fullWidth
+          onClick={() => {
+            // TODO throw a modal
+            submitFormWithStatus(false, ListingStatus.changesRequested)
+          }}
+        >
+          {t("listings.approval.requestChanges")}
+        </Button>
+      </GridCell>
+    )
 
-      if (listing.events.find((event) => event.type === ListingEventType.lotteryResults)) {
-        const eventUrl = pdfUrlFromListingEvents(
-          listing?.events,
-          ListingEventType.lotteryResults,
-          process.env.cloudinaryCloudName
+    const reopenButton = (
+      <GridCell key="btn-reopen">
+        <Button
+          id="publishButton"
+          styleType={AppearanceStyleType.success}
+          type="button"
+          fullWidth
+          onClick={() => {
+            // TODO throw a modal
+            submitFormWithStatus(true, ListingStatus.active)
+          }}
+        >
+          {t("listings.approval.reopen")}
+        </Button>
+      </GridCell>
+    )
+
+    const getApprovalActions = () => {
+      const elements = []
+      // read-only form
+      if (type === "details") {
+        if (isListingApprover) {
+          // admins can approve and publish if pending approval or changes requested
+          if (
+            listing.status === ListingStatus.pendingReview ||
+            listing.status === ListingStatus.changesRequested
+          )
+            elements.push(approveAndPublishButton)
+          // admins can always edit
+          elements.push(editFromDetailButton)
+        } else {
+          // partners cannot edit if pending approval
+          if (listing.status !== ListingStatus.pendingReview) elements.push(editFromDetailButton)
+        }
+
+        // all users can preview
+        elements.push(previewButton)
+
+        // all users can view lottery results if posted
+        if (listing.events.find((event) => event.type === ListingEventType.lotteryResults)) {
+          const eventUrl = pdfUrlFromListingEvents(
+            listing?.events,
+            ListingEventType.lotteryResults,
+            process.env.cloudinaryCloudName
+          )
+          elements.push(viewPostedResultsButton(eventUrl))
+        }
+      }
+
+      // new unsaved listing
+      if (type === "add") {
```

Comment:
For "type" could you make the `ListingFormActionsType` an enum instead of the three strings? That way you could use the enum in these if statements?